### PR TITLE
Prevent LombokValToFinalVar from removing star imports

### LINKED ISSUE #962: UpgradeToJava21 removes `import lombok.*;` without adding explicit imports
## What version of OpenRewrite are you using?

- OpenRewrite plugin v7.23.0
- rewrite-recipe-bom v3.4.0
- rewrite-migrate-java (from BOM)

## How are you running OpenRewrite?

Gradle plugin via init script on a multi-module project (~200 modules).

```groovy
initscript {
    repositories {
        maven { url "https://plugins.gradle.org/m2" }
    }
    dependencies { classpath("org.openrewrite:plugin:7.23.0") }
}
rootProject {
    plugins.apply(org.openrewrite.gradle.RewritePlugin)
    dependencies {
        rewrite(platform("org.openrewrite.recipe:rewrite-recipe-bom:3.4.0"))
        rewrite("org.openrewrite.recipe:rewrite-migrate-java")
    }
    rewrite {
        activeRecipe("org.openrewrite.java.migrate.UpgradeToJava21")
    }
}
```

## What is the smallest, simplest way to reproduce the problem?

```java
package com.example;

import lombok.*;

@AllArgsConstructor
@NoArgsConstructor
@ToString
@Data
@EqualsAndHashCode
public class AuthHeaders {
    String authHeader;
    String token;
}
```

Run: `./gradlew --init-script init.gradle rewriteRun`

Note: The bug occurs in multi-module projects. I could not reproduce it in a single-module project.

## What did you expect to see?

Either unchanged code, or expanded explicit imports:

```java
package com.example;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;

@AllArgsConstructor
@NoArgsConstructor
@ToString
@Data
@EqualsAndHashCode
public class AuthHeaders {
    String authHeader;
    String token;
}
```

## What did you see instead?

The `import lombok.*;` line was removed without replacement:

```java
package com.example;

@AllArgsConstructor
@NoArgsConstructor
@ToString
@Data
@EqualsAndHashCode
public class AuthHeaders {
    String authHeader;
    String token;
}
```

This breaks compilation because the Lombok annotations are no longer imported.

## Scale of impact

In our project, `rewriteDryRun` reports:
- 95 files lose `import lombok.*;`
- 0 files gain explicit lombok imports

The recipe chain shows:
```
org.openrewrite.java.migrate.UpgradeToJava21
  org.openrewrite.java.migrate.UpgradeToJava17
    org.openrewrite.java.migrate.Java8toJava11
      org.openrewrite.java.migrate.lombok.UpdateLombokToJava11
        org.openrewrite.java.migrate.lombok.LombokValToFinalVar
```

## What is the full stack trace of any errors you encountered?

No stack trace. The recipe completes successfully but produces invalid code.

## Are you interested in contributing a fix to OpenRewrite?

Yes, with guidance on where the import removal logic resides.

### PR DESCRIPTION
**Pull request by Jenson3210**
## Summary

- Only call `maybeRemoveImport("lombok.var")` when there is an explicit `import lombok.var;`, not from star imports like `import lombok.*;`
- Added tests for star import preservation with and without var usage, explicit import removal, and edge cases

## Problem

Running `UpgradeToJava21` on multi-module Gradle projects removes `import lombok.*;` wildcard imports without adding explicit replacements, breaking compilation. The reporter observed 95 files losing their star import with 0 files gaining explicit lombok imports.

The root cause is in `LombokValToFinalVar` which unconditionally called `maybeRemoveImport("lombok.var")` in `visitCompilationUnit`. 

## Solution

Changed `visitCompilationUnit` to check the compilation unit's import list: only call `maybeRemoveImport("lombok.var")` when there is an explicit `import lombok.var;` statement (where `getQualid().getSimpleName()` is not `"*"`). Star imports are left untouched.

Note: when `import lombok.*;` exists solely for `var` (no other lombok usage), the star import will remain as an unused import after the recipe runs. This is an acceptable trade-off—an unused import is preferable to broken compilation.

## Test plan

- [x] Existing tests pass
- [x] `preserveStarImportWithoutVarUsage` — star import + annotations, no var, incomplete type info
- [x] `preserveStarImportWithVarUsage` — star import + var + annotations, incomplete type info
- [x] `removeExplicitVarImport` — explicit `import lombok.var;` removed with incomplete type info
- [x] `removeStarImportWhenOnlyValUsed` — star import correctly removed when val is the only lombok usage
- [x] `starImportRemainsWhenOnlyVarUsed` — star import remains unused (documented trade-off)

- Fixes #962

### CODE REVIEW

File: src/test/java/org/openrewrite/java/migrate/lombok/LombokValToFinalVarTest.java [hunk 1]
```diff
@@ -113,6 +114,141 @@ void bar() {
...
+          )
+        );
+    }
+
+    @Test
+    void starImportRemainsWhenOnlyVarUsed() {
+        // When `import lombok.*;` exists only for `var`, the star import remains unused after
+        // the recipe runs. This is acceptable: removing a star import with incomplete type info
+        // (e.g. in multi-module projects) risks breaking compilation by dropping imports that
+        // other lombok types still need. An unused import is preferable to broken code.
```
Comment (Jenson3210) [inline src/test/java/org/openrewrite/java/migrate/lombok/LombokValToFinalVarTest.java:231]:
```suggestion
        // When `import lombok.*;` exists only for `var`, the star import remains unused after
        // the recipe runs. This is acceptable: removing a star import with incomplete type info
        // risks breaking compilation by dropping imports that
        // other lombok types still need. An unused import is preferable to broken code.
```

File: src/main/java/org/openrewrite/java/migrate/lombok/LombokValToFinalVar.java [hunk 2]
```diff
@@ -61,7 +61,17 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
     private static class LombokValToFinalVarVisitor extends JavaIsoVisitor<ExecutionContext> {
         @Override
         public J.CompilationUnit visitCompilationUnit(J.CompilationUnit compilationUnit, ExecutionContext ctx) {
-            maybeRemoveImport(LOMBOK_VAR);
+            // Only remove explicit `import lombok.var;`, not from star imports like `import lombok.*;`
+            // which may also cover other lombok types still in use. With incomplete type info
+            // (e.g. in multi-module projects), maybeRemoveImport on a star import can incorrectly
+            // remove the entire import.
```
Comment (Jenson3210) [inline src/main/java/org/openrewrite/java/migrate/lombok/LombokValToFinalVar.java]:
```suggestion
```

File: src/main/java/org/openrewrite/java/migrate/lombok/LombokValToFinalVar.java [hunk 3]
```diff
@@ -61,7 +61,13 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
...
         public J.CompilationUnit visitCompilationUnit(J.CompilationUnit compilationUnit, ExecutionContext ctx) {
-            maybeRemoveImport(LOMBOK_VAR);
+            for (J.Import imp : compilationUnit.getImports()) {
+                if (!imp.isStatic() && LOMBOK_VAR.equals(imp.getTypeName()) &&
+                        !"*".equals(imp.getQualid().getSimpleName())) {
+                    maybeRemoveImport(LOMBOK_VAR);
+                    break;
+                }
+            }
             return super.visitCompilationUnit(compilationUnit, ctx);
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/migrate/lombok/LombokValToFinalVar.java:71]: Would this be easier perhaps than manually inspecting the imports?
```suggestion
            J.CompilationUnit cu = super.visitCompilationUnit(compilationUnit, ctx);
            if (cu != compilationUnit) {
                maybeRemoveImport(LOMBOK_VAR);
            }
            return cu;
```

Comment (Jenson3210) [reply src/main/java/org/openrewrite/java/migrate/lombok/LombokValToFinalVar.java:71]: Can't do this as star imports would be removed causing the bug to reappear
